### DISCUSSION TITLE
Running both node & python in the same micro

### DISCUSSION THREAD

**Discussion by Abban-Fahim** [deta/cloud-docs#424] [Help]
Basically I have a **node js server** running express (works fine) that runs a **python file** on the server and generates some files that I then serve. I use the `child_process.exec` method from node (not a npm library, its a built in node js libary). 

I previously used to host this server on Heroku, where I could change my project setting to have multiple runtimes (but only one entry point, `index.js` in my case). Is there any way I could perform similar tasks on a deta micro? Perhaps by changing something in `.deta/proj_info`?

**Comment by jnsougata** [comment]
I can't answer your query exactly but you can run a python script in a js micro as all micros are x64 linux lambdas, so python is builtin. there should be a os lib in js from which you can run `python3 your_script.py` in cli

**Reply by Abban-Fahim** [reply]
Yes that os lib is `child_process.exec()`. however python3, py, and python are all `command not found` in the cli. I debugged this by trying different commands and then returning the error output as json. I also looked through the list of all commands available in the bash cli using `compgen -c`. Looking through the 400+ commands available in the cli - node, npm & npx, are all avaialable but not python.

Is there a way I could perhaps install a package only once upon deployment using the command line, via a package manager? Do the macros even have a package manager like apt-get or pacman
